app/src/main/java/com/example/bigowlapp/viewModel/MonitoringGroupPageViewModel.java

```diff
@@ -52,4 +53,14 @@ private void loadAllUsersInGroup(Group group) {
         usersInGroup = userRepository.getDocumentsByListOfUId(group.getSupervisedUserId(), User.class);
     }
 
+    public void removeUserFromGroup(User userToBeRemoved) {
+        Group groupWithRemovedUser = getGroup().getValue();
+        Objects.requireNonNull(groupWithRemovedUser).getSupervisedUserId().remove(userToBeRemoved.getUId());
```

Comment:
groupWithRemovedUser should be in an if (groupWithRemovedUser != null) with a throw exception, or a try-catch clause. This doesn't handle an exception.

Reply:
Not an issue; it's already handled. Forget about it.

Reply:
The method can only be called when a user can actually first see the monitoring group with a list of users, perform a long press click on any of the user, and click the "remove" button. Hence, no check is required for method removeUserFromGroup because the activity page and ViewModel already checked before a supervisor can even remove a user.